// rc-spinner/dist/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _objectAssign = require('object-assign');

var _objectAssign2 = _interopRequireDefault(_objectAssign);

var _spin = require('./spin');

var _spin2 = _interopRequireDefault(_spin);

var _reactDom = require('react-dom');

var _reactDom2 = _interopRequireDefault(_reactDom);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Spin = function (_Component) {
  (0, _inherits3.default)(Spin, _Component);

  function Spin() {
    (0, _classCallCheck3.default)(this, Spin);
    return (0, _possibleConstructorReturn3.default)(this, (0, _getPrototypeOf2.default)(Spin).apply(this, arguments));
  }

  (0, _createClass3.default)(Spin, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      var el = _reactDom2.default.findDOMNode(this);
      var props = this.props;
      this.spin = (0, _spin2.default)(el, {
        color: props.color,
        duration: props.duration,
        width: props.lineWidth
      });
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      if (this.spin) this.spin.stop();
    }
  }, {
    key: 'render',
    value: function render() {
      var props = this.props;
      var style = (0, _objectAssign2.default)(props.style || {}, {
        width: props.height,
        height: props.height
      });
      return _react2.default.createElement(
        'div',
        { className: props.className,
          style: style },
        props.children
      );
    }
  }]);
  return Spin;
}(_react.Component);

Spin.defaultProps = {
  color: '#cccccc',
  height: 30,
  duration: 1000,
  lineWidth: 4
};
Spin.propTypes = {
  color: _propTypes2.default.string,
  height: _propTypes2.default.number,
  duration: _propTypes2.default.number,
  lineWidth: _propTypes2.default.number
};
exports.default = Spin;
